specification/protocol/otlp.md

```diff
@@ -145,16 +148,47 @@ was not delivered.
 
 #### OTLP/gRPC Response
 
-The server may respond with either a success or an error to the requests.
+The response MUST be the appropriate serialized Protobuf message (see below for
+the specific message to use in the [Success](#success),
+[Partial Success](#partial-success) and [Failure](#failures) cases).
+
+##### Success
 
 The success response indicates telemetry data is successfully processed by the
 server. If the server receives an empty request (a request that does not carry
 any telemetry data) the server SHOULD respond with success.
 
-Success response is returned via
-[Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto)
-message (`ExportTraceServiceResponse` for traces, `ExportMetricsServiceResponse`
-for metrics, `ExportLogsServiceResponse` for logs).
+On success, the server response MUST be
+a Protobuf-encoded [Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto) message
+(`ExportTraceServiceResponse` for traces,
+`ExportMetricsServiceResponse` for metrics and
+`ExportLogsServiceResponse` for logs).
+
+The server SHOULD respond with success no sooner than after successfully
+decoding and validating the request.
+
+The server MUST leave the `partial_success` field unset
+in case of a successfull response.
+
+##### Partial Success
+
+If the processing of the request is partially successfull
+(i.e. when the server accepts only parts of the data and rejects the rest) the
+server MUST respond with the same message types as specified on the Success case.
+
+Additionally, the server MUST initialize the `partial_success` field
+(`ExportTracePartialSuccess` message for traces,
+`ExportMetricsPartialSuccess` message for metrics and
+`ExportLogsPartialSuccess` message for logs), and it MUST set the respective
+`accepted_spans`, `accepted_data_points` or `accepted_log_records` field with
+the number of spans/data points/log records it accepted.
+
+The server MAY populate the `error_message` field with a human-readable
+error message in English. The protocol does not attempt to define the content
+nor structure of such error message, but it SHOULD generally offer guidance
```

Comment:
```suggestion
nor structure of such error message, but it should generally offer guidance
```
I don't think this should be an uppercase should. If its uppercase, there is some semantics associated with it, and this statement is pretty vague.

Reply:
I think its fine that its vague, but I don't think we should attach semantic meaning to it

Reply:
Makes sense to remove the should. The vagueness was on purpose :D